handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -796,6 +796,11 @@ public final SSLEngineResult unwrap(
             }
 
             int packetLength = SslUtils.getEncryptedPacketLength(srcs, srcsOffset);
+
+            if (packetLength == SslUtils.NOT_ENCRYPTED) {
```

Comment:
@normanmaurer - can you provide a link? From what I can tell it looks like the JDK does this check up front.

http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/8b04ee324a1a/src/share/classes/sun/security/ssl/SSLEngineImpl.java#l868 -> http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/8b04ee324a1a/src/share/classes/sun/security/ssl/EngineInputRecord.java#l156

Also it doesn't make sense to report underflow/overflow if the data isn't legitimate.